src/current/cockroachcloud/upgrade-to-v24.2.md

```diff
@@ -0,0 +1,296 @@
+{{ support_ending_soon }}
+{{site.data.alerts.end}}
+    {% elsif purple_result < eos %}
+{{site.data.alerts.callout_version}}
+{{ support_ending_soon }}
+{{site.data.alerts.end}}
+    {% endif %}
+  {% endif %}
+{% else %}
+{{ testing_release_message }}
+{% endif %}
+
+{% capture previous_version_numeric %}{{ x.major_version | remove_first: 'v' }}{% endcapture %}
+{% capture major_version_numeric %}{{ x.previous_version | remove_first: 'v' }}{% endcapture %}
+
+{% if DEBUG == true %}
+previous_version_numeric: {{ previous_version_numeric }}<br />
+major_version_numeric: {{ major_version_numeric }}<br />
+{% endif %}
+
+{% if page.pre_production_preview == true and released == false %}
+[CockroachDB {{ page.pre_production_preview_version }}](https://www.cockroachlabs.com/docs/releases/{{ x.major_version }}#{{ page.pre_production_preview_version | replace: ".","-"}}) is available to CockroachDB {{ site.data.products.dedicated }} clusters as an opt-in upgrade for testing and experimentation.
+
+{{site.data.alerts.callout_danger}} [Testing releases]({% link releases/index.md %}#release-naming) are not qualified for production environments and not eligible for support or uptime SLA commitments.
+{{site.data.alerts.end}}
+
+An [Org Administrator]({% link cockroachcloud/authorization.md %}#org-administrator) can upgrade your CockroachDB {{ site.data.products.dedicated }} cluster from the CockroachDB {{ site.data.products.cloud }} Console. This page shows how to upgrade a CockroachDB {{ site.data.products.dedicated }} cluster to {{ page.pre_production_preview_version }} for testing and experimentation.
+
+{{site.data.alerts.callout_success}}
+Upgrading from {{ x.previous_version }} to {{ page.pre_production_preview_version }} is a major-version upgrade. Upgrading a CockroachDB {{ site.data.products.dedicated }} cluster to a new major version is opt-in. Before proceeding, review the CockroachDB {{ site.data.products.cloud }} [CockroachDB Cloud Upgrade Policy](https://cockroachlabs.com/docs/cockroachcloud/upgrade-policy#pre-production-preview). After a cluster is upgraded to a Pre-Production Preview release, it is automatically upgraded to all subsequent releases within the same major version—including additional beta and RC releases, the GA release, and subsequent patch releases after GA, as patch version upgrades. To learn more, refer to [Patch Version Upgrades]({% link cockroachcloud/upgrade-policy.md %}#patch-version-upgrades).
+{{site.data.alerts.end}}
+
+{% elsif released == true %}
+
+Now that [CockroachDB {{ x.major_version }}](https://www.cockroachlabs.com/docs/releases/{{ x.major_version }}) is available, an [Org Administrator]({% link cockroachcloud/authorization.md %}#org-administrator) can upgrade your CockroachDB {{ site.data.products.dedicated }} cluster from the [CockroachDB {{ site.data.products.cloud }} Console](https://cockroachlabs.cloud). This page shows how to upgrade a cluster in CockroachDB Cloud to {{ x.major_version }}.
```

Comment:
this is not building a working link, it's making this link:

```
https://www.cockroachlabs.com/docs/releases/%20v24.2
```
I think removing the space before the `x` in the following will fix it

```
.../releases/{{ x.major_version }} ...
```